File: problems/problem_002/javascript/solution.test.js
import { romanToInt } from './solution';
import { test, expect, describe } from 'bun:test';

describe('romanToInt', () => {
    const tests = [
        ['XLII', 42],
        ['CXCIV', 194],
        ['MMMCDXLIV', 3444],
    ];

    test.each(tests)(
        'for roman string = %s, should return %d',
        (romanValue, expected) => {
            expect(romanToInt(romanValue)).toEqual(expected);
        }
    );
});

File: problems/problem_002/golang/solution_test.go
package problem_002

import "testing"

type romanToIntegerTest struct {
	s        string
	expected int
}

var romanToIntegerTests = []romanToIntegerTest{
	{"XLII", 42},
	{"CXCIV", 194},
	{"MMMCDXLIV", 3444},
}

func Test_romanToInteger(t *testing.T) {
	for _, test := range romanToIntegerTests {
		if result := romanToInteger(test.s); result != test.expected {
			t.Errorf("Output %v not equal to expected %v", result, test.expected)
		}
	}
}
